compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaTypeReferenceTypeSymbol.java

```diff
@@ -166,4 +169,8 @@ public String signature() {
     private boolean isAnonOrg(ModuleID moduleID) {
         return ANON_ORG.equals(moduleID.orgName());
     }
+
+    private boolean isAnonType(String definitionName) {
+        return definitionName != null && definitionName.startsWith("$anonType$");
```

Comment:
What if we ensure to check for `$` symbol? If in any case we change the anon type symbol name this will not be valid. Also, we should do the same for internal symbol as well. WDYT?

Reply:
We are using the same check in BallerinaClassSymbol as well. 
But the suggestion makes sense, will check and update.